internal/controller/pipelinerollout_controller.go

```diff
@@ -545,7 +558,7 @@ func (r *PipelineRolloutReconciler) processPipelineStatus(ctx context.Context, p
 	}
 
 	// Get existing Pipeline
-	existingPipelineDef, err := kubernetes.GetCR(ctx, r.restConfig, pipelineDef, "pipelines")
+	existingPipelineDef, err := kubernetes.GetLiveResource(ctx, pipelineDef, "pipelines")
```

Comment:
Hmm, we need to think about what to do here. If we call `GetLiveResource()` every time here, then we are effectively making an API call with every reconciliation of the Pipeline. I kind of wonder if we should stop doing any kind of "Get" here and simply get the Pipeline from the previous `reconcile()` call and pass it into here ?

Reply:
Instead of getting it from previous `reconcile()`, how about getting it from cache instead? because getting from previous `reconsile` is neither or less same as getting from cache.
I ran the unit test with it and don't see any failure.

Reply:
Yaa, it might be a problem!

Reply:
Updated

Reply:
After change, Unit test and E2E is started getting NPE, I tried to look but didn't found the root issue in time-bound, So thinking to look it as a separate issue. what do you think?

Reply:
Do you mean to leave the code making the K8S API call here for now? And then doing a follow up to avoid making the K8S call?

Reply:
Yes, But still one Unit test case if failing, which i'm fixing

Reply:
okay, let me know when ready

Reply:
and just add a "TODO" here to indicate that it will be changed to not call the K8S API

Reply:
Fixed the test cases